sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -535,18 +542,78 @@ private[client] class Shim_v2_0 extends Shim with Logging {
     alterPartitionsMethod.invoke(hive, tableName, newParts)
   }
 
-  override def dropPartition(
+  override def dropPartitions(
       hive: Hive,
       dbName: String,
       tableName: String,
-      part: JList[String],
+      specs: Seq[TablePartitionSpec],
       deleteData: Boolean,
+      ignoreIfNotExists: Boolean,
       purge: Boolean): Unit = {
     val dropOptions = new PartitionDropOptions
     dropOptions.deleteData(deleteData)
     dropOptions.purgeData(purge)
+    dropOptions.ifExists(ignoreIfNotExists)
     recordHiveCall()
-    hive.dropPartition(dbName, tableName, part, dropOptions)
```

Comment:
a switch is needed to allow user to restore the previous behavior

Reply:
Add a new switch named `spark.sql.dropPartitionInBatch.enabled`.